src/main/java/github/scarsz/discordsrv/objects/threads/PlayingStatusUpdater.java

```diff
@@ -0,0 +1,119 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package github.scarsz.discordsrv.objects.threads;
+
+import alexh.weak.Dynamic;
+import github.scarsz.discordsrv.DiscordSRV;
+import github.scarsz.discordsrv.objects.Lag;
+import github.scarsz.discordsrv.util.DiscordUtil;
+import github.scarsz.discordsrv.util.MemUtil;
+import github.scarsz.discordsrv.util.PlayerUtil;
+import github.scarsz.discordsrv.util.TimeUtil;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.Bukkit;
+
+import java.io.File;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class PlayingStatusUpdater extends Thread {
+    public PlayingStatusUpdater() {
+        setName("DiscordSRV - Status Updater");
+    }
+
+    private int lastStatus = 0;
+    @Override
+    public void run() {
+        while (true) {
+            int rate;
+            // Breaking change to config
+            try {
+                rate = DiscordSRV.config().getInt("StatusUpdateRateInMinutes");
+            } catch (IllegalArgumentException e) {
+                DiscordSRV.error("\"StatusUpdaterRateInMinutes\" not found in config");
+                rate=2;
+            }
+
+            if (rate < 1) rate = 1;
+
+            if (DiscordUtil.getJda() != null) {
+
+                Dynamic dynamic = DiscordSRV.config().dget("DiscordGameStatus");
+                List<String> statuses = new LinkedList<>();
+                if (dynamic.isList()) {
+                    statuses.addAll(dynamic.asList());
+                } else {
+                    statuses.add(dynamic.convert().intoString());
+                }
+                String status = applyPlaceholders(statuses.get(lastStatus));
+
+                // Increment and wrap around
+                lastStatus++;
+                if (lastStatus == statuses.size() - 1)
+                    lastStatus = 0;
+
+                if (!StringUtils.isEmpty(status))
+                    DiscordUtil.setGameStatus(status);
+                else
+                    DiscordSRV.debug("Skipping status update cycle, status was null");
+            } else {
+                DiscordSRV.debug("Skipping status update cycle, JDA was null");
+            }
+
+            try {
+                Thread.sleep(TimeUnit.MINUTES.toMillis(rate));
+            } catch (InterruptedException e) {
+                DiscordSRV.error("Broke from Status Updater thread: sleep interrupted");
+                return;
+            }
+        }
+    }
+
+    @Getter private static File playerDataFolder = new File(Bukkit.getWorlds().get(0).getWorldFolder().getAbsolutePath(), "/playerdata");
+    @SuppressWarnings({"SpellCheckingInspection", "ConstantConditions"})
+    private static String applyPlaceholders(String input) {
```

Comment:
This is a direct copy of https://github.com/DiscordSRV/DiscordSRV/blob/75d61bc51d522e2b85320253dfc60bac0586d2d2/src/main/java/github/scarsz/discordsrv/objects/threads/ChannelTopicUpdater.java#L67 (should be 1 method, not 2. Bad practice)